dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/deduplication/PotentialDuplicatesEnrollmentsTests.java

```diff
@@ -0,0 +1,233 @@
+            .body( "events", hasSize( greaterThanOrEqualTo( 1 ) ) )
+            .body( "events.dataValues", hasSize( greaterThanOrEqualTo( 1 ) ) )
+            .noRootPath().rootPath( String.format( "enrollments.find{it.enrollment=='%s'}", enrollmentB ) )
+            .body( "events", hasSize( greaterThanOrEqualTo( 1 ) ) )
+            .body( "events.dataValues", hasSize( greaterThanOrEqualTo( 1 ) ) );
+
+        trackerActions.getTrackedEntity( teiB )
+            .validate().statusCode( 404 );
+    }
+
+    @Test
+    public void shouldMergeWithNonSuperUser()
+    {
+        // arrange
+        String teiB = createTeiWithoutEnrollment();
+
+        TrackerApiResponse imported = trackerActions.postAndGetJobReport( trackerActions
+            .buildTeiWithEnrollmentAndEvent( Constants.ORG_UNIT_IDS[0], Constants.TRACKER_PROGRAM_ID, "nlXNK4b7LVr" ) )
+            .validateSuccessfulImport();
+
+        String teiA = imported.extractImportedTeis().get( 0 );
+        String enrollment = imported.extractImportedEnrollments().get( 0 );
+        String potentialDuplicate = potentialDuplicatesActions.createAndValidatePotentialDuplicate( teiA, teiB, "OPEN" );
+
+        String username = createUserWithAccessToMerge();
+        new LoginActions().loginAsUser( username, userPassword );
+
+        // act
+        potentialDuplicatesActions.autoMergePotentialDuplicate( potentialDuplicate ).validate().statusCode( 200 );
+
+        // assert
+        potentialDuplicatesActions.get( potentialDuplicate )
+            .validate()
+            .statusCode( 200 )
+            .body( "status", equalTo( "MERGED" ) );
+
+        trackerActions.getTrackedEntity( teiA + "?fields=*" )
+            .validate().statusCode( 200 )
+            .body( "enrollments", hasSize( 1 ) )
+            .body( "enrollments.enrollment", hasItems( enrollment ) )
+            .body( String.format( "enrollments.find{it.enrollment=='%s'}.events", enrollment ), hasSize( 1 ) );
+
+        trackerActions.getTrackedEntity( teiB ).validate().statusCode( 404 );
+    }
+
+    private TrackerApiResponse createTeiWithEnrollmentsAndEvents( String program, String programStage )
+    {
+        return trackerActions.postAndGetJobReport( trackerActions
+            .buildTeiWithEnrollmentAndEvent( Constants.ORG_UNIT_IDS[0], program, programStage ) )
+            .validateSuccessfulImport();
+    }
+
+    private String createTeiWithoutEnrollment()
+    {
+        JsonObject object = trackerActions.buildTei( Constants.TRACKED_ENTITY_TYPE, Constants.ORG_UNIT_IDS[0] );
+
+        return trackerActions.postAndGetJobReport( object ).extractImportedTeis().get( 0 );
+    }
+
+    private String createUserWithAccessToMerge()
+    {
+
+        String username = DataGenerator.randomString();
+        String auth = new UserRoleActions().createWithAuthorities( "F_TRACKED_ENTITY_MERGE" );
+        String userid = new UserActions().addUser( username, userPassword );
+
```

Comment:
also F_TRACKED_ENTITY_MERGE, could we have a common library shared between projects?

Reply:
E2E tests shouldn't have dependencies on the projects we are testing or reuse any code. That's because in case of a change in authority name, tests wouldn't fail, but in reality, we would be breaking the clients.